### PR TITLE
add mmfashion dataset

### PR DESCRIPTION
**Pull request by veralauee**
add mmfashion dataset for cloth detection and segmentation

### CODE REVIEW

File: mmdet/datasets/mmfashion.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from .builder import DATASETS
+from .coco import CocoDataset
+
+
+@DATASETS.register_module()
+class mmfashionDataset(CocoDataset):
```
Comment (hellock) [inline mmdet/datasets/mmfashion.py:6]: Rename it to `DeepFashionDataset`?

File: configs/mmfashion/mask_rcnn_r50_fpn_1x.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+_base_ = [
```
Comment (hellock) [inline configs/mmfashion/mask_rcnn_r50_fpn_1x.py:1]: Follow the [naming conventions](https://mmdetection.readthedocs.io/en/latest/config.html#config-name-style) to name the config file.

File: configs/mmfashion/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+## MMFashion
```
Comment (hellock) [inline configs/mmfashion/README.md:1]: Add a blank line after the H2 heading.

File: mmdet/datasets/mmfashion.py (hunk 1 above)
Comment (veralauee) [reply mmdet/datasets/mmfashion.py]: Renamed mmfashion to DeepFashion in the new commit.

File: configs/mmfashion/mask_rcnn_r50_fpn_1x.py (hunk 2 above)
Comment (veralauee) [reply configs/mmfashion/mask_rcnn_r50_fpn_1x.py]: Renamed the config file following the naming conventions in the new commit.

File: configs/mmfashion/README.md (hunk 3 above)
Comment (veralauee) [reply configs/mmfashion/README.md]: added.

File: configs/deepfashion/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+```
+
+After that you can train the Mask RCNN r50 on DeepFashion-In-shop dataset by launching training with the `mask_rcnn_r50_fpn_1x.py` config 
+or creating your own config file.
+
+## Model Zoo
+|   Backbone  |  Model type  |       Dataset       |  bbox detection Average Precision  | segmentation Average Precision |      Download (Google)      |
+| :---------: | :----------: | :-----------------: | :--------------------------------: | :----------------------------: | :-------------------------: |
+|   ResNet50  |   Mask RCNN  | DeepFashion-In-shop |                0.599               |              0.584             |  [model](https://drive.google.com/open?id=1q6zF7J6Gb-FFgM87oIORIt6uBozaXp5r)   |
```
Comment (hellock) [inline configs/deepfashion/README.md:43]: We need to upload both the model and log.

Comment (veralauee) [reply configs/deepfashion/README.md]: log uploaded.

File: configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    '../_base_/models/mask_rcnn_r50_fpn.py',
+    '../_base_/datasets/deepfashion.py', '../_base_/schedules/schedule_1x.py',
+    '../_base_/default_runtime.py'
+]
+model = dict(
+    roi_head=dict(
+        bbox_head=dict(num_classes=15), mask_head=dict(num_classes=15)))
+# runtime settings
+total_epochs = 12
+workflow = [('train', 5)]
```
Comment (hellock) [inline configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py:11]: `('train', 5)` and `('train', 1)` are actually the same.

File: configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+_base_ = [
+    '../_base_/models/mask_rcnn_r50_fpn.py',
+    '../_base_/datasets/deepfashion.py', '../_base_/schedules/schedule_1x.py',
+    '../_base_/default_runtime.py'
+]
+model = dict(
+    roi_head=dict(
+        bbox_head=dict(num_classes=15), mask_head=dict(num_classes=15)))
+# runtime settings
+total_epochs = 12
```
Comment (hellock) [inline configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py:10]: This is the default setting? It may be omitted from this config.

Comment (veralauee) [reply configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py]: Delete the unnecessary settings. 
roi_head is different due to different num_classes.

Comment (hellock) [reply configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py]: It seems that you changed total_epochs from 12 to 15.

Comment (veralauee) [reply configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py]: Yes. 15 epochs is better-performed, so I updated the config, checkpoint and the log.

Comment (hellock) [reply configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py]: If so, you may update the filename from `1x` to `15e`.

Comment (veralauee) [reply configs/deepfashion/mask_rcnn_r50_fpn_1x_deepfashion.py]: Renamed. Thanks for your advice.

**Comment by openmmlab-v** [comment]
Hi @veralauee ！First of all, we want to express our gratitude for your significant PR in the mmdetection project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you are Chinese or have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**